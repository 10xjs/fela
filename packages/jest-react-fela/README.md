# Jest React Fela

<img alt="npm version" src="https://badge.fury.io/js/jest-react-fela.svg"> <img alt="npm downloads" src="https://img.shields.io/npm/dm/jest-react-fela.svg">

Testing Tools for [Jest]() in combination with [react-fela](../react-fela).

## Installation
```sh
yarn add jest-react-fela
```
You may alternatively use `npm i --save jest-react-fela`.

## Usage
This package exposes a single API that creates a snapshot.<br>
Check the [Usage Guide]() and [API Reference]() for more information.


## License
Fela is licensed under the [MIT License](http://opensource.org/licenses/MIT).<br>
Documentation is licensed under [Creative Common License](http://creativecommons.org/licenses/by/4.0/).<br>
Created with â™¥ by [@rofrischmann](http://rofrischmann.de) and all the great contributors.
