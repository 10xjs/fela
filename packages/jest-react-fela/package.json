{
  "name": "jest-react-fela",
  "version": "10.0.0",
  "description": "Jest helpers for react-fela",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "files": [
    "LICENSE",
    "README.md",
    "lib/**",
    "es/**"
  ],
  "repository": "https://github.com/rofrischmann/fela/",
  "keywords": [
    "fela",
    "react-fela",
    "testing",
    "react",
    "jest",
    "jest-utils"
  ],
  "author": "Robin Frischmann",
  "license": "MIT",
  "peerDependencies": {
    "fela": "*",
    "react": "*",
    "react-dom": "*",
    "react-fela": "*"
  },
  "dependencies": {
    "jest-fela-bindings": "^10.0.0"
  },
  "devDependencies": {
    "fela": "^10.0.0",
    "fela-preset-web": "^10.0.0",
    "react-fela": "^10.0.0"
  }
}
