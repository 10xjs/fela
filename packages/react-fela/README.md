# React Fela

<img alt="npm version" src="https://badge.fury.io/js/react-fela.svg"> <img alt="npm downloads" src="https://img.shields.io/npm/dm/react-fela.svg">

Official [React](https://github.com/facebook/react) bindings for Fela.

This package only includes React bindings for [Fela](http://github.com/rofrischmann/fela).<br>
It assumes you already know about Fela and how to use it.

## Installation
```sh
yarn add react-fela
```
You may alternatively use `npm i --save react-fela`.

## Usage
For more information and best practices on how to effectively use this package, please check out the main [Fela Documention - Usage with React](http://fela.js.org/docs/guides/UsageWithReact.html). You will find some useful examples and a basic introduction on how to use Fela with React.<br>
Checkout the [API Reference - Bindings](http://fela.js.org/docs/API.html#bindings) for further technical information on how the React/Preact/Inferno bindings work.

## Migration
Since version 10.0.0 there have been some major changes in the APIs. Here you'll find some useful migration guides in order to upgrade.
x
- [connect → FelaComponent](migration/connect.md)
- [createComponent → FelaComponent](migration/createComponent.md)
- [withTheme → FelaTheme](migration/withTheme.md)

## License
Fela is licensed under the [MIT License](http://opensource.org/licenses/MIT).<br>
Documentation is licensed under [Creative Common License](http://creativecommons.org/licenses/by/4.0/).<br>
Created with ♥ by [@rofrischmann](http://rofrischmann.de) and all the great contributors.
