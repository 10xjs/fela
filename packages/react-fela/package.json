{
  "name": "react-fela",
  "version": "10.1.3",
  "description": "React bindings for Fela",
  "typings": "index.d.ts",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "files": [
    "LICENSE",
    "README.md",
    "index.d.ts",
    "lib/**",
    "es/**"
  ],
  "repository": "https://github.com/rofrischmann/fela/",
  "keywords": [
    "fela",
    "dynamic styling",
    "react",
    "css",
    "styling",
    "cssinjs"
  ],
  "author": "Robin Frischmann",
  "license": "MIT",
  "peerDependencies": {
    "react": "^16.3.0"
  },
  "dependencies": {
    "fela-bindings": "^10.1.3",
    "fela-dom": "^10.1.3",
    "prop-types": "^15.5.8"
  }
}
