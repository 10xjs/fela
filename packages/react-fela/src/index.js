/* @flow */
import connect from './connect'
import createComponent from './createComponent'
import createComponentWithProxy from './createComponentWithProxy'
import FelaComponent from './FelaComponent'
import FelaTheme from './FelaTheme'
import RendererProvider from './RendererProvider'
import ThemeProvider from './ThemeProvider'
import withTheme from './withTheme'
import fe from './fe'

import { interceptDeprecation } from './_deprecate'

const Provider = interceptDeprecation(
  RendererProvider,
  'Importing `Provider` from react-fela is deprecated. Import `RendererProvider` instead.\nSee http://fela.js.org/docs/api/bindings/renderer-provider'
)

export {
  connect,
  createComponent,
  createComponentWithProxy,
  FelaComponent,
  FelaTheme,
  Provider,
  RendererProvider,
  ThemeProvider,
  withTheme,
  fe,
}
