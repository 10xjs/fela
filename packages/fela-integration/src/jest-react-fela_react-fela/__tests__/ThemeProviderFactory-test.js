import 'raf/polyfill'
import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { createSnapshot } from 'jest-react-fela'
import { ThemeProvider, FelaTheme } from 'react-fela'

describe('Using the ThemeProvider', () => {
  it('should pass the theme to rule props', () => {
    const snapshot = createSnapshot(
      <ThemeProvider theme={{ color: 'red' }}>
        <FelaTheme>{theme => <div>{JSON.stringify(theme)}</div>}</FelaTheme>
      </ThemeProvider>
    )

    expect(snapshot).toMatchSnapshot()
  })
})
