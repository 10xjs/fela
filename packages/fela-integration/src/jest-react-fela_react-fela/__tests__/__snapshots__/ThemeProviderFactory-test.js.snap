// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using the ThemeProvider should pass the theme to rule props 1`] = `
"

<div>
  {{}color:red{}}
</div>;
"
`;
